src/python/zquantum/core/wip/circuit/conversions/symbolic/qiskit_expressions.py

```diff
@@ -0,0 +1,71 @@
+"""Translations between Qiskit parameter expressions and intermediate expression trees.
+
+Attributes:
+    QISKIT_DIALECT: Mapping from the intermediate expression tree into atoms
+        used in Qiskit symbolic expressions. Allows translating an expression
+        into the Qiskit dialect. Can be used with
+        `zquantum.core.circuit.symbolic.translations.translate_expression`.
+"""
+import operator
+from functools import reduce, singledispatch
+from numbers import Number
+
+import qiskit
+
+from .expressions import ExpressionDialect
+from .helpers import reduction
+from .sympy_expressions import expression_from_sympy
+
+
+@singledispatch
+def expression_from_qiskit(expression):
+    """Parse Qiskit expression into intermediate expression tree."""
+    raise NotImplementedError(
+        f"Expression {expression} of type {type(expression)} is currently not supported"
+    )
+
+
+@expression_from_qiskit.register
+def _number_identity(number: Number):
+    return number
+
+
+@expression_from_qiskit.register
+def _expr_from_qiskit_param_expr(
+    qiskit_expr: qiskit.circuit.parameterexpression.ParameterExpression,
+):
+    # At the moment of writing this (qiskit==0.16.1) there's no other way to introspect
```

Comment:
Looks like nah, I still couldn't find a way to do it with Qiskit's public API.

I've updated the docstring. I'm using `0.23.2` because it's a strict `==` constraint in our `setup.py`.

Reply:
After discussing with @alexjuda we decided to mention here that it is not possible to introspect symbolic expression in the version we of Qiskit depend on as well as in the newest version. Done in 53eaaaf.

Reply:
Sounds good, thx!